// logger-sf/lib/models/Log.js
const DEFAULT_ENV = "staging";
const DEFAULT_COMMERCE = "Seguros"
const DEFAULT_LOG_LEVEL = "info"
class Log {

    constructor() {
        this.commerce = process.env.COMMERCE || DEFAULT_COMMERCE;
        this.country = process.env.COUNTRY ;
        this.date = new Date().toISOString();
        this.environment = process.env.ENVIRONMENT || DEFAULT_ENV;
        this.details = undefined;
        this.level = process.env.LOG_LEVEL || DEFAULT_LOG_LEVEL;
        this.message = undefined;
        this.product = undefined;
        this.traceID = undefined;
    }

    withCommerce (commerce = process.env.COMMERCE || DEFAULT_COMMERCE) {
        this.commerce = commerce;
        return this;
    }

    withCountry (country = process.env.COUNTRY) {
        this.country = country;
        return this;
    }

    withEnvironment (environment = process.env.ENVIRONMENT || DEFAULT_ENV) {
        this.environment = environment;
        return this;
    }

    withDetails (details = '') {
        this.details = details;
        return this;
    }

    withLevel (level = process.env.LOG_LEVEL || DEFAULT_LOG_LEVEL) {
        this.level = level;
        return this;
    }


    withMessage (message = '') {
        this.message = message;
        return this;
    }

    withProduct (product = '') {
        this.product = product;
        return this;
    }

    withTraceID (traceID = '') {
        this.traceID = traceID;
        return this;
    }
}

module.exports = Log;
